music_assistant/common/models/media_items.py

```diff
@@ -318,16 +319,17 @@ def __hash__(self) -> int:
         """Return custom hash."""
         return hash(self.uri)
 
-    def __eq__(self, other: MediaItem | ItemMapping) -> bool:
+    def __eq__(self, other: object) -> bool:
         """Check equality of two items."""
+        if not isinstance(other, MediaItem | ItemMapping):
+            return False
         return self.uri == other.uri
 
 
 @dataclass(kw_only=True)
 class MediaItem(_MediaItemBase):
     """Base representation of a media item."""
 
-    __hash__ = _MediaItemBase.__hash__
```

Comment:
I didn't know why these were explicitly assigned like this rather than using the inherited ones, but they break the "self" typing so I had to remove a few.

Reply:
Otherwise they dont work - they do not inherit otherwise

Reply:
This way breaks mypy because it expects self to be MediaItemBase rather than MediaItem, so i'll have to add a function that just calls super() i think.

Reply:
I have no idea why it doesn't work though - could be a bug in python 3.12 or it is simply not meant to inherit

Reply:
4d5e2e47ac58d9cb1986754201662b0bcd0d3cb4